meta/engineering_principles.md

```diff
@@ -0,0 +1,189 @@
+
+- Further Reading
+
+  - [The Artsy Support Process](http://artsy.github.io/blog/2018/05/25/support-process/)
+  - [Introducing Peril to Artsy](http://artsy.github.io/blog/2017/09/04/Introducing-Peril/)
+  - [Artsy Apogee](http://artsy.github.io/blog/2018/02/02/artsy-apogee/)
+
+### Leverage Your Impact
+
+We built an large reputation for our size as an engineering team via the other principles. This gives internal
+engineers the ability to move local impact to a larger communal impact.
+
+Consider ways in which you can expand the scope of impact for your time:
+
+- Avoid private messaging when you could communicate in a public channel
+- Team presentations could be re-worked for public consumption
+- Internal notes could be turned into blog posts
+- Documentation on teams and processes could be open for others to reference and learn from
+- A regular meetup could be converted into a franchisable system for others to replicate
+
+The small improvements eventually add up so that you can do industry level work as a part of building Artsy.
+
+- Examples
+
+  - [Artsy x React Native](http://artsy.github.io/artsy-x-react-native.html)
+  - [JavaScriptures](http://artsy.github.io/series/javascriptures/)
+  - [Learn Swift at Artsy](http://artsy.github.io/series/swift-at-artsy/)
+  - Artsy Blog
+  - [artsy/meta][meta]
+  - [Peer Labs](https://peerlab.community)
+  - [Artsy Engineering Compensation Framework](http://artsy.github.io/blog/2015/04/03/artsy-engineering-compensation-framework/)
+
+### De-silo Engineers
+
+Provide people the ability to work in many spaces and not feel blocked by another team. Common languages, common
+terminology and common idioms. There's definitely space for experimentation inside a shared framework. The aim here
+is not to stifle creativity, but increase individual scope. By consolidating and de-siloing you can put concentrate
+your efforts into maximizing value for time.
+
+We intentionally try to avoid teams/people to be silo'd from an organizational perspective by allowing ourselves to
+re-staff periodically throughout a year.
+
+- Examples
+
+  - Moving to React Native
+  - Teach Swift via Swift at Artsy
+  - Migrating Admin tools to shared Front-End stacks
+  - Consolidated Analytics schema
+  - Dev Swap which switches teams and responsibilities
+
+- Further Reading
+  - [Loosely Held Strong Convictions](https://www.youtube.com/watch?v=hlLhtWLghGA)
+  - [DevSwap: An Experiment in Randomizing Teams](http://artsy.github.io/blog/2017/09/11/DevSwap/)
+  - [Artsy x React Native](http://artsy.github.io/artsy-x-react-native.html)
+
+### Build for 10x
+
+A system won't live forever, but you should make decisions with a aim to handle 10x the traffic, number of
+contributors and company revenue. Basically a single order of magnitude of growth everywhere. This can help guide
+deciding on technological choices and figuring the trade-offs between feature work and infrastructure needs.
```

Comment:
My attempt at re-wording this (take what you like):

> Technology choices should strive to be "optimal" while avoid over-engineering. When designing systems or evaluating scalability and performance, we aim for today's decisions to withstand 10x the traffic, data, or scale. This realistic horizon helps us balance the need to move quickly with the sometimes-competing need to invest in infrastructure and architecture. It also recognizes that solutions are expected to evolve and be replaced.

Reply:
I love it, replaced.